lib/scsb-client.js

```diff
@@ -0,0 +1,29 @@
+const SCSBClient = require('@nypl/scsb-rest-client')
+const kmsHelper = require('./kms-helper')
+
+let _instance = null
+
+const instance = async () => {
+  if (!_instance) {
```

Comment:
The first time we process a batch of several items, we're going to call `instance()` for each of those items. Because creds will not have yet been decrypted, each of those calls to `instance()` will enter this `if` block, and proceed to 1) decrypt `SCSB_URL`, followed by 2) decrypt `SCSB_API_KEY`. Ideally, a bunch of simultaneous calls to `instance()` would not _each_ attempt to decrypt the same sets of credentials; There's a way to ensure that the very first of them fires off the decryption calls and all of the runners up await the same kms result:

```
const instance = async () => {
  if (!_instance) {
    // Do "preflight check" here...
    
    // Decrypt url and key _simultaneously_ because they don't depend on each other:
    _instance = Promise.all([
      kmsHelper.decrypt("url..."),
      kmsHelper.decrypt("key...")
    ])  
      .then((creds) => {
        const [ decryptedUrl, decryptedKey ] = creds

        _instance = new SCSBClient({
          url: decryptedUrl,
          apiKey: decryptedKey
        })  

        return _instance
      })  
  }

  return _instance2
}
```

So, the first call to `instance()` will find that `_instance` is falsey and set it to a `Promise`, which will eventually resolve a client. If another call to `instance()` is made at any point before the first call has resolved, the second call will  simply get the existing `Promise`. So if we're processing 1000 items on that first run, we'll only decrypt creds once and all of the 1000 calls to `instance()` will get the same resolved scsb client at the same time.